internal/scm/gerrit/gerrit.go

```diff
@@ -56,35 +55,50 @@ func New(username, token, baseURL, repoSearch string) (*Gerrit, error) {
 }
 
 func (g Gerrit) GetRepositories(ctx context.Context) ([]scm.Repository, error) {
-	opt := &gogerrit.ProjectOptions{
-		Description: true,
-		Regex:       g.repoSearch,
-		Type:        "CODE",
-		ProjectBaseOptions: gogerrit.ProjectBaseOptions{
-			Limit: 2500, // Maybe we should make this configurable
-		},
-	}
-	projects, _, err := g.client.ListProjects(ctx, opt)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to list projects")
-	}
-
 	repos := make([]scm.Repository, 0)
-	for _, name := range slices.Sorted(maps.Keys(*projects)) {
-		project := (*projects)[name]
-		if project.State != "ACTIVE" {
-			log.Debug("Skipping repository since state is not ACTIVE")
-			continue
+	skip := 0
+	for {
```

Comment:
There is an opportunity to break things out of this outer for to make things easier to read. I let you descide if you want to do it or not.

Reply:
Can't make the suggestion on partyally deleted lines. But take everything within the outer `for {` into its own function 


```go
func (g Gerrit) getRepositoriesWithSkip(ctx context.Context, skip int) (repos []scm.Repository, moreProjects bool err error) {
```
(Might have missed something necessary)

And call out to it. Will make it easier to read the padding logic, and will make it easier to read the querying logic :)